### PR TITLE
docs: Add example(s) with `Project(..., mode="mlflow")`

### PR DESCRIPTION
**Pull request by cakedev0**
#### Change description

Add an example to the gallery for the MLFlow backend, link: https://skoredoc.eks.probabl.dev/2532/auto_examples/technical_details/plot_mlflow_backend.html

The examples is simple:
- create a CV report
- put it in MLFlow
- ~~screenshot of what it looks like on MLFlow~~ => removed as per offline discussions with @auguste-probabl (judged as not the good place and hard to maintain, more adequate for communication on LinkedIn for instance).
- show link between summarize id and MLFlow run ID.

#### Contribution checklist

- [x] The code passes our style conventions (you can check this by running pre-commit on your code
with `pre-commit run --all-files`)
- [x] All the tests pass (please test locally before pushing)
- [x] :question: The documentation builds and renders properly (if it does, our bot will add a comment linking
to a preview of the documentation to review it visually)
- [x] All the commits in the PR are signed (more information
[here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits))
- [x] The pull request title respects the Conventional Commits convention (more information
[here](https://docs.skore.probabl.ai/dev/contributing.html#pull-requests))

#### AI usage disclosure

AI tools were involved for: research and understanding + the initial draft, but I rewrote it entirely.

### CODE REVIEW

File: .github/workflows/sphinx.yml [hunk 1]
```diff
@@ -230,7 +230,11 @@ jobs:
 
       - name: Install
         working-directory: skore
-        run: python -m pip install --force-reinstall --no-deps .
+        run: |
+          set -eu
+          python -m pip install --force-reinstall --no-deps .
+          python -m pip install --upgrade "mlflow==3.10.0"
+          python -m pip install --force-reinstall --no-deps ../skore-mlflow-project
```
Comment (thomass-dev) [inline .github/workflows/sphinx.yml]: Reference: https://github.com/probabl-ai/skore/pull/2527#discussion_r2863598002.

Comment (cakedev0) [reply .github/workflows/sphinx.yml]: Fixed.

File: examples/technical_details/plot_mlflow_backend.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    # example page focuses on skore usage rather than backend startup details.
+    with redirect_stdout(io.StringIO()), redirect_stderr(io.StringIO()):
+        project = Project(
+            PROJECT,
+            mode="mlflow",
+            tracking_uri=TRACKING_URI,
+        )
+        # This creates an MLflow experiment with name `PROJECT`
+
+    project.put("hgb-baseline", report)
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]: This shows up as indented in the rendered docs which looks a bit weird. I would say that if the `try` block fails, the only consequence is some temp files don't get cleaned up which is not so bad.

File: examples/technical_details/plot_mlflow_backend.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+==================================
+MLflow backend with skore.Project
+==================================
+
+This example shows how to persist reports in MLflow using
+:class:`~skore.Project` in ``mode="mlflow"``: log reports
+as MLflow runs and inspect them.
+
+Examples
+--------
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]: This title doesn't really make sense?

File: examples/technical_details/plot_mlflow_backend.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    # MLflow/Alembic emits verbose DB initialization logs; silence them so the
+    # example page focuses on skore usage rather than backend startup details.
+    with redirect_stdout(io.StringIO()), redirect_stderr(io.StringIO()):
+        project = Project(
+            PROJECT,
+            mode="mlflow",
+            tracking_uri=TRACKING_URI,
+        )
+        # This creates an MLflow experiment with name `PROJECT`
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]: `PROJECT` is defined in the `ignore` block so it's weird to mention it here

File: examples/technical_details/plot_mlflow_backend.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+try:
+    # sphinx_gallery_end_ignore
+
+    import io
+    from contextlib import redirect_stderr, redirect_stdout
+
+    # MLflow/Alembic emits verbose DB initialization logs; silence them so the
+    # example page focuses on skore usage rather than backend startup details.
+    with redirect_stdout(io.StringIO()), redirect_stderr(io.StringIO()):
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]: There are still some warnings printed, even with this. I wonder if we could tell sphinx gallery not to show the output of the cell at all?

File: examples/technical_details/plot_mlflow_backend.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+finally:
+    if tmp_dir is not None:
+        tmp_dir.cleanup()
+
+# sphinx_gallery_end_ignore
+
+
+# %%
+# You get warnings about serialization, future versions of `skore[mlflow]` might remove
+# those warnings by using `skops.io` for models serialization.
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]:
```suggestion
# Note that mlflow warns us about saving models with `pickle`. Future versions of skore might rely on
# `skops <https://skops.readthedocs.io>`__ for model serialization, which will make these warnings disappear.
```

File: examples/technical_details/plot_mlflow_backend.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+# Like for other types of projects (local, hub), you can access the summary
+# and its DataFrame version:
+import pandas as pd
+
+summary = project.summarize()
+pandas_summary = pd.DataFrame(summary)
+pandas_summary[["key", "report_type", "learner", "ml_task", "dataset"]]
+
+# %%
+# The "id" is the MLflow run ID, you can access the MLflow run this way if you want:
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]:
```suggestion
# The "id" column corresponds to the MLflow run ID, so you can access the MLflow run this way:
```

File: examples/technical_details/plot_mlflow_backend.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+import mlflow
+
+_, run_id = pandas_summary.index[0]
+
+mlflow_run = mlflow.get_run(run_id)
+mlflow_run.data.metrics
+
+# %%
+# But most importantly, you can load back a project using this id:
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]:
```suggestion
# But most importantly, this ID lets you load saved reports:
```

File: examples/technical_details/plot_mlflow_backend.py (hunk 4 above)
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]: Move this comment above `project = `

File: examples/technical_details/plot_mlflow_backend.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+Examples
+--------
+
+To run this example and push in your own MLflow tracking server, you can run
+this example with the following command:
+
+.. code-block:: bash
+
+    TRACKING_URI=<tracking_uri> PROJECT=<project> python plot_skore_hub_project.py
```
Comment (auguste-probabl) [inline examples/technical_details/plot_mlflow_backend.py]:
```suggestion
    TRACKING_URI=<tracking_uri> PROJECT=<project> python plot_mlflow_backend.py
```

File: examples/technical_details/plot_mlflow_backend.py (hunk 5 above)
Comment (auguste-probabl) [reply examples/technical_details/plot_mlflow_backend.py]: Oh actually we address the remaining warnings later. Hm. I think I'd still rather try to hide all the cell outputs.

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+# those warnings by using `skops.io` for models serialization.
+
+# %%
+# Like for other types of projects (local, hub), you can access the summary
+# and its DataFrame version:
+import pandas as pd
+
+summary = project.summarize()
+pandas_summary = pd.DataFrame(summary)
+pandas_summary[["key", "report_type", "learner", "ml_task", "dataset"]]
```
Comment (auguste-probabl) [inline examples/technical_details/plot_skore_mlflow_project.py]: maybe?
```suggestion
pandas_summary = pd.DataFrame(summary).set_index("id")
pandas_summary[["key", "report_type", "learner", "ml_task", "dataset"]]
```

Comment (auguste-probabl) [reply examples/technical_details/plot_skore_mlflow_project.py]: Do we really need to select the columns? It makes the example a bit less elegant

Comment (cakedev0) [reply examples/technical_details/plot_skore_mlflow_project.py]: Hum "id" is already the index I think? (given how it's )

I think selecting columns makes the output quite cleaner: otherwise you have a super wide dataframe, which creates a horizontal scrollbar.

File: examples/technical_details/plot_mlflow_backend.py (hunk 9 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: Ooops! thx

File: examples/technical_details/plot_mlflow_backend.py (hunk 2 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: Ok, I hesitated. I'll remove the `try/finally` then.

File: examples/technical_details/plot_mlflow_backend.py (hunk 6 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: Nice thanks.

File: examples/technical_details/plot_mlflow_backend.py (hunk 3 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: Nope, removed it.

File: examples/technical_details/plot_mlflow_backend.py (hunk 4 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: I think it's fine because this example start with :

> To run this example and push in your own MLflow tracking server, you can run
> this example with the following command:
> `TRACKING_URI=<tracking_uri> PROJECT=<project> python plot_skore_mlflow_project.py`

The hub example has the same pattern.

File: examples/technical_details/plot_mlflow_backend.py (hunk 5 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: I can put the `Project.put` in the `with` block, but we still have to write a comment about those specific logs so that users are not surprised to se them once they try it on their side.

File: examples/technical_details/plot_mlflow_backend.py (hunk 4 above)
Comment (cakedev0) [reply examples/technical_details/plot_mlflow_backend.py]: Good point, done.

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+# sphinx_gallery_start_ignore
+if tmp_dir is not None:
+    tmp_dir.cleanup()
+
+# sphinx_gallery_end_ignore
+
+
+# %%
+# Note that mlflow warns us about saving models with `pickle`. Future versions of skore might rely on
+# `skops <https://skops.readthedocs.io>`__ for model serialization, which will make these warnings disappear.
```
Comment (glemaitre) [inline examples/technical_details/plot_skore_mlflow_project.py]: Could you wrap 88 chars.

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+# Like for other types of projects (local, hub), you can access the summary
+# and its DataFrame version:
+import pandas as pd
+
+summary = project.summarize()
+pandas_summary = pd.DataFrame(summary)
+pandas_summary[["key", "report_type", "learner", "ml_task", "dataset"]]
+
+# %%
+# The "id" column corresponds to the MLflow run ID, so you can access the MLflow run this way:
```
Comment (glemaitre) [inline examples/technical_details/plot_skore_mlflow_project.py]: This line is a bit long.

**Comment by glemaitre** [comment]
I just realized that we are not dropping the artifact for the train and test set but only the test set. We should probably do both (it could be done in another PR).

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+====================
+MLflow skore Project
+====================
+
+This example shows how to persist reports in MLflow using
+:class:`~skore.Project` in ``mode="mlflow"``: log reports
+as MLflow runs and inspect them.
+
+To run this example and push in your own MLflow tracking server, you can run
+this example with the following command:
```
Comment (glemaitre) [inline examples/technical_details/plot_skore_mlflow_project.py]: I would suggest what a user would need to do to launch the MLflow server and execute the example to provide more guidance.

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+    TRACKING_URI=<tracking_uri> PROJECT=<project> python plot_skore_mlflow_project.py
+
+"""
+
+# %%
+# Build one report to persist
+# ===========================
+
+from sklearn.datasets import load_iris
```
Comment (auguste-probabl) [inline examples/technical_details/plot_skore_mlflow_project.py:27]:
```suggestion
# %%
# First, let us build one report to persist:

# %%
from sklearn.datasets import load_iris
```

File: examples/technical_details/plot_skore_mlflow_project.py [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+X, y = load_iris(return_X_y=True, as_frame=True)
+
+estimator = HistGradientBoostingClassifier()
+report = CrossValidationReport(estimator, X, y)
+
+# %%
+# Push the report to the MLflow backend
+# =====================================
+
+# sphinx_gallery_start_ignore
```
Comment (auguste-probabl) [inline examples/technical_details/plot_skore_mlflow_project.py:41]:
```suggestion
# %%
# Then, we can push the report to the MLflow backend:

# %%
# sphinx_gallery_start_ignore
```

File: examples/technical_details/plot_skore_mlflow_project.py (hunk 10 above)
Comment (auguste-probabl) [reply examples/technical_details/plot_skore_mlflow_project.py]: Yeah so i'd like `id` to be the *only* index. Note the `0` before the id:

Image

or maybe put id as a normal column, to avoid the vertical space between the column names and the table.


ok for the columns subset

File: examples/technical_details/plot_skore_mlflow_project.py (hunk 13 above)
Comment (cakedev0) [reply examples/technical_details/plot_skore_mlflow_project.py]: Good point. Showed how to do that with `uvx` (probably the easiest way).

**Review by thomass-dev — APPROVED** [review]

**Comment by cakedev0** [comment]
I'm not sure exactly why, but running `bash pip-compile.sh --sphinx-requirements --upgrade` broke everything, I had to updates the `pyproject.toml`s and then re-run it to get the sphinx build back up.

@thomass-dev Can you take a last look please? Esp. do you think the changes to the `pyproject.toml`s are ok?

**Comment by thomass-dev** [comment]
> Esp. do you think the changes to the pyproject.tomls are ok?

Yes, totally! Let me investigate why the `--upgrade` is wrong.

**Comment by cakedev0** [comment]
I think what was wrong was not requiring a minimal `mlflow` version in `skore-mlflow-project`. Though, I'm not sure why previous lock files were working correctly :thinking:

**Comment by thomass-dev** [comment]
> I think what was wrong was not requiring a minimal `mlflow` version in `skore-mlflow-project`. Though, I'm not sure why previous lock files were working correctly 🤔

After a bunch of tests, it looks like `pandas` and `skore-mlflow-project` are conflicting (i don't know why), and `uv` finds a solution with `mlflow==1.27.0`. I will continue my investigation.

**Comment by thomass-dev** [comment]
```
dependencies = [                
  "pandas",                     
  "skore-mlflow-project>=0.0.2",
]

-> mlflow==1.27.0
```

```
dependencies = [                
  "skore-mlflow-project>=0.0.2",
  "pandas",                     
]

-> mlflow==3.10.1
```

https://docs.astral.sh/uv/pip/compatibility/#package-priority

**Comment by thomass-dev** [comment]
I've just put a commit to force uv/pip to upgrade `mlflow` b46f22d834a7774d174ce63a87c9a84f3de1ebb5.
Let's merge it as is and see what happens after the release of `skore-mlflow-project`.

**Review by thomass-dev — APPROVED** [review]
